packages/zsa-react/src/index.ts

```diff
@@ -238,7 +251,9 @@ export const useServerAction = <
 
   let final: TServerActionResult<TServerAction>
 
-  if (isExecuting && oldResult.status === "empty") {
+  const isPending = isTransitioning || isExecuting
```

Comment:
`isPending` should be both `isTransitioning` and `isExecuting` because on retries during the delay `isTransitioning` would be false and `isExecuting` will remain true.